types/es-abstract/scripts/collect-intrinsics.ts

```diff
@@ -0,0 +1,742 @@
+#!/usr/bin/env ts-node --script-mode
+/// <reference types="node"/>
+import path = require('path');
+import fs = require('fs');
+
+const OUT_FILE_PATH = path.resolve(__dirname, '..', 'GetIntrinsic.d.ts');
+
+const hasSymbols = true;
+const { getPrototypeOf: getProto, getOwnPropertyDescriptor: $gOPD } = Reflect as {
+    getPrototypeOf(target: object): any;
+    getOwnPropertyDescriptor<T extends object, P extends PropertyKey>(
+        target: T,
+        propertyKey: P,
+    ): (P extends keyof T ? TypedPropertyDescriptor<T[P]> : PropertyDescriptor) | undefined;
+};
+
+function isObject(value: unknown): value is object {
+    if (value === undefined || value === null) return false;
+    return typeof value === 'object' || typeof value === 'function';
+}
+
+// tslint:disable: only-arrow-functions space-before-function-paren
+const ThrowTypeError = (function () {
+    return $gOPD(arguments, 'callee')!.get;
+})();
+
+// tslint:disable: no-async-without-await
+const generator = function* () {};
+const asyncFn = async function () {};
+const asyncGen = async function* () {};
+// tslint:enable
+
+const generatorFunction = generator ? /** @type {GeneratorFunctionConstructor} */ generator.constructor : undefined;
+const generatorFunctionPrototype = generatorFunction ? generatorFunction.prototype : undefined;
+const generatorPrototype = generatorFunctionPrototype ? generatorFunctionPrototype.prototype : undefined;
+
+const asyncFunction = asyncFn ? /** @type {FunctionConstructor} */ asyncFn.constructor : undefined;
+
+const asyncGenFunction = asyncGen ? /** @type {AsyncGeneratorFunctionConstructor} */ asyncGen.constructor : undefined;
+const asyncGenFunctionPrototype = asyncGenFunction ? asyncGenFunction.prototype : undefined;
+const asyncGenPrototype = asyncGenFunctionPrototype ? asyncGenFunctionPrototype.prototype : undefined;
+
+const TypedArray: any = typeof Uint8Array === 'undefined' ? undefined : getProto(Uint8Array);
```

Comment:
why does this need `any`?

Reply:
Because `TypedArray` doesn’t have type definitions in the **TypeScript** standard library.

**TypeScript** defines those methods on the individual **TypedArray** subclasses in the type definitions.

Reply:
ok, then wouldn't it be a union of the subclasses?

Reply:
Because that has its own issues, most notably that **TypeScript** wouldn’t be able to infer method and property types correctly.